github.go

```diff
@@ -235,6 +240,23 @@ func (g *GitHubClient) GetZipballLink(tag string) (*url.URL, error) {
 	return u, nil
 }
 
+func FilterReleasesByTagName(releases []*github.RepositoryRelease, tagNameRegex string) ([]*github.RepositoryRelease, error) {
```

Comment:
I agree, it was made public for testing so that the fakes could actually filter by regex, if we change the approach in the fake then we can switch this to private.